client/scripts/build-modules.js

```diff
@@ -0,0 +1,186 @@
+const { nodeExternalsPlugin } = require("esbuild-node-externals")
+const { readdirSync, realpathSync } = require("fs")
+const { performance } = require("perf_hooks")
+const browserslist = require("browserslist")
+
+// -- CONSTANTS, COMMAND LINE ARGUMENTS
+
+const [opts] = cliopts.parse(
+  ["self", "Compiles the calling module package only."],
+  ["tests", "Compile the module tests instead."]
+)
+
+const DEV = Boolean(cliopts.watch)
+const SELF = opts.self ?? false
+const TESTS = opts.tests ?? false
+
+if (DEV && TESTS) throw new Error("Can't use -watch and -tests at the same time!")
+
+const fmt = TESTS ? styl.orange : styl.blue
+
+/** Common settings for all modules that are built. */
+const SETTINGS_COMMON = {
+  // esbuild settings
+  outdir: "dist",
+  // minify: true // estrella implicitly toggles minify depending on the `--debug` flag
+  bundle: true,
+  treeShaking: true,
+  splitting: true,
+  format: "esm",
+  platform: "browser",
+  sourcemap: true,
+  sourcesContent: true,
+
+  // estrella settings
+  tslint: false, // disables estrella's built-in tsc typechecking
+  quiet: true // silences estrella's logging, as we use our own logging
+}
+
+// prettier-ignore
+const DEMANGLE_TARGET_REGEX =
+  /^(chrome|and_(?:ff|chr)|edge|firefox|ios_saf|safari)\s+([\d.]+)/
+
+// -- DEMANGLE BROWSERSLIST
+
+/*
+  esbuild expects the browsers list provided to it look like this:
+  chrome89, firefox78, etc.
+  Browserslist looks like this:
+  chrome 89, firefox 78, etc.
+  Additionally, esbuild only accepts certain browsers.
+  So, this chunk of code converts the browserslist list into an esbuild list.
+  It also renames some of the mobile browsers to match esbuild's version.
+*/
+
+const targets = new Set()
+for (const target of browserslist()) {
+  const result = DEMANGLE_TARGET_REGEX.exec(target)
+  if (!result) continue
+  let [, browser, version] = result
+  // rewrite mobile browser versions into what esbuild understands
+  if (browser === "ios_saf") browser = "ios"
+  else if (browser === "and_chr") browser = "chrome"
+  else if (browser === "and_ff") browser = "firefox"
+  targets.add(browser + version)
+}
+
+// -- BUILD
+
+// check if we're building all or just a single package
+if (SELF) buildModule()
+else directoriesOf("./modules").forEach(name => buildModule(name))
+
+// -- FUNCTIONS
+
+/** Builds the module (a package in `modules/`).
+ *  If `-tests` was specified, it'll compile the module's tests, instead. */
+function buildModule(name) {
+  let dir, index, package, tests
+
+  if (SELF) {
+    dir = process.cwd()
+    index = "./src/index.ts"
+    package = "./package.json"
+  } else {
+    dir = realpathSync(`./modules/${name}`)
+    index = `${dir}/src/index.ts`
+    package = `${dir}/package.json`
+  }
+
+  if (TESTS) {
+    try {
+      tests = filesOf(`${dir}/tests`).map(name => `./tests/${name}`)
+      if (tests.length === 0) {
+        console.log(fmt(`[${name}]`), "No tests, skipping.")
+        return
```

Comment:
You're gonna hate me for this
```suggestion
        throw new Error()
```

Reply:
How would I possibly test `node-snowpack-workers`? That's something that requires something more akin to a browser test, which we have not figured out how to do at all.

Reply:
The line below this suggestion is a `catch` that emits the exact same console.log

Reply:
oh right. it's 4am. I don't see the need to throw an error, and then have to catch it, when you can just return out of the function. plus, the two code paths mean two separate things even if they emit the same log